File: tellme/js/app/NoteService.js
var NoteService = app.service('NoteService', [ '$http', 'config', function($http, config) {
	this.getPublicRootNotes = function() {
	    return $http({
	        method : 'GET',
	        url : config.serverUrl + 'note/get_public_root_notes',
	    });
	}
	
	this.getNote = function( noteId) {
	    return $http({
	        method : 'GET',
	        url : config.serverUrl + 'note/get_note/'+noteId,
	    });
	}
	
	this.addNote = function addNote( note) {
	    return $http({
	        method : 'POST',
	        url : config.serverUrl + 'note/add_note',
	        data : note
	    });
	}
	
	this.getUserNote = function getUserNote(userId) {
	    return $http({
	        method : 'GET',
	        url : config.serverUrl + 'note/get_user_note/'+userId,
	    });
	}

	this.addQuestion = function(quest) {
	    return $http({
	        method : 'POST',
	        url : config.serverUrl + 'note/add_question',
	        data : quest
	    });
	}
	
	this.addKeyPoint = function( kpoint) {
	    return $http({
	        method : 'POST',
	        url : config.serverUrl + 'note/add_keypoint',
	        data : kpoint
	    });
	}
	
} ]);

File: note-module/notemodule/src/main/java/microservices/tellme/notemodule/service/NoteService.java
package microservices.tellme.notemodule.service;

import java.util.List;

import microservices.tellme.notemodule.domain.KeyPoint;
import microservices.tellme.notemodule.domain.Note;
import microservices.tellme.notemodule.domain.Question;

public interface NoteService {
	
	List<Note> getPublicRootNotes();
	
	String saveOrUpdateNote(Note note);
	
	String saveOrUpdateQuestion(Question question);
	
	String saveOrUpdateKeyPoint(KeyPoint keyPoint);
	
	Note getUserNote(String userId);
	
	Note getNote(String noteId);
	
	List<Note> getChildNotes(Long noteId);
	
	List<Question> getQuestions(Long noteId);
	
	List<KeyPoint> getKeyPoints(Long questionId);
	

}
